conjureup/ui/views/applicationconfigure.py

```diff
@@ -25,45 +25,50 @@ def __init__(self, application, close_cb):
         super().__init__()
 
     def build_widget(self):
+        app.loop.create_task(self._build_widget())
+        return []
+
+    async def _build_widget(self):
         ws = []
         num_unit_ow = OptionWidget("Units", "int",
                                    "How many units to deploy.",
-                                   self.application.orig_num_units,
+                                   self.application.num_units,
                                    current_value=self.num_units_copy,
                                    value_changed_callback=self.handle_scale)
         ws.append(num_unit_ow)
-        ws += self.get_whitelisted_option_widgets()
+        ws += await self.get_whitelisted_option_widgets()
         self.toggle_show_all_button_index = len(ws) + 1
         self.toggle_show_all_button = SecondaryButton(
             "Show Advanced Configuration",
-            self.do_toggle_show_all_config)
+            lambda sender: app.loop.create_task(
+                self.do_toggle_show_all_config()))
```

Comment:
Might be nice to add some logic to `SecondaryButton` to detect coroutines and do the right thing, so that we don't have to deal with this all over the place.